ppmclibs/tinycv_impl.cc

```diff
@@ -197,11 +181,15 @@ std::vector<int> search_TEMPLATE(const Image* scene, const Image* object,
     int scene_width = scene_bottom_x - scene_x;
     int scene_height = scene_bottom_y - scene_y;
 
-    Mat scene_copy = scene->prep();
-    Mat object_copy = object->prep();
+    Mat scene_roi = Mat(scene->img, Rect(scene_x, scene_y, scene_width, scene_height));
+    Mat object_roi = Mat(object->img, Rect(x, y, width, height));
 
-    Mat scene_roi(scene_copy, Rect(scene_x, scene_y, scene_width, scene_height));
-    Mat object_roi(object_copy, Rect(x, y, width, height));
+    for (auto t : { &scene_roi, &object_roi }) {
+        Mat n;
+        GaussianBlur(*t, n, Size(3, 3), 0, 0);
```

Comment:
The OpenCV behaviour is somewhat unintuitive when it comes to borders and ROIs:
http://docs.opencv.org/3.2.0/d2/de8/group__core__array.html#ga209f2f4869e304c82d07739337eae7c5

By default (i.e. if BORDER_ISOLATED is not specified), the blur will use the original values, i.e. the blurred ROI will end up exactly as before.

This actually threw a test error when doing the cvtColor(...) before the blur, as it created a new Mat with the size of the ROI. You can try it if you replace the *t in the GaussianBlur with t->clone().